gluten-core/src/main/scala/io/glutenproject/expression/UnaryExpressionTransformer.scala

```diff
@@ -118,8 +118,37 @@ class CheckOverflowTransformer(
         Seq(original.dataType, BooleanType),
         FunctionConfig.OPT))
 
+    // just make a fake toType value, because native engine cannot accept datatype itself
+    val toTypeNodes = ExpressionBuilder.makeDecimalLiteral(
+      new Decimal().set(0, original.dataType.precision, original.dataType.scale))
     val expressionNodes =
-      Lists.newArrayList(childNode, new BooleanLiteralNode(original.nullOnOverflow))
+      Lists.newArrayList(childNode, toTypeNodes, new BooleanLiteralNode(original.nullOnOverflow))
```

Comment:
Why do you need to pass the parameter `toTypeNodes`,  there is already a return type in the substrait plan.

Reply:
@jinchengchenghh Can you help to check whether we can use the return type directly in the substrait plan?

Reply:
This is from `CheckOverflow` constructor, velox should use it to match function signature, and decide if it is `ShortDecimal` or `LongDecimal`, so as `MakeDecimal`. This is a special case, the input argument is same with return type, but I think we should not do some exceptional thing when convert substrait plan to velox plan

Reply:
@JkSelf  @jinchengchenghh Maybe we can do this special logic only when backend is velox, or just append `toTypeNodes` after `nullOnOverflow` instead before it.  Currently CH didn't need `toTypeNodes`,  those changes would cause an incompatiability issue as in CH uts.

Reply:
OK, I will move it after `nullOnOverflow`

Reply:
Please add comment 'test by CH[[378]]'.
https://github.com/Kyligence/ClickHouse/pull/378

Reply:
I have moved it, but CH backend still has error
```
java.lang.RuntimeException: check_overflow function requires two args.
```

Reply:
Can you merge this one https://github.com/Kyligence/ClickHouse/pull/378?  So this PR can pass CH CI